### PR TITLE
Gradle tweaks

### PR DESCRIPTION
**Pull request by 3flex**
Just rearranging a few things to make the build config a little more idiomatic, as well as fix a deprecation warning and fix a couple of issues in the README examples.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -51,9 +39,3 @@ tasks {
         }
     }
 }
-
-apiValidation {
-    // We need to perform api validations for external APIs, for :detekt-api and :detekt-psi-utils
-    ignoredProjects.addAll(subprojects.filter { it.name !in listOf("detekt-api", "detekt-psi-utils") }.map { it.name })
-    ignoredPackages.add("io.gitlab.arturbosch.detekt.api.internal")
-}
```
Comment (BraisGabin) [inline build.gradle.kts:59]: 👏

**Review by BraisGabin — APPROVED** [review]
great refactor!

File: detekt-api/build.gradle.kts [hunk 2]
```diff
@@ -30,3 +31,7 @@ listOf(configurations.testFixturesApiElements, configurations.testFixturesRuntim
 tasks.withType<DokkaTask>().configureEach {
     outputDirectory.set(rootDir.resolve("docs/pages/kdoc"))
 }
+
+apiValidation {
+    ignoredPackages.add("io.gitlab.arturbosch.detekt.api.internal")
```
Comment (schalkms) [inline detekt-api/build.gradle.kts:36]: Isn't this necessary for `detekt-psi-utils` as well?

**Review by schalkms — APPROVED** [review]

File: detekt-api/build.gradle.kts (hunk 2 above)
Comment (3flex) [reply detekt-api/build.gradle.kts:36]: The `io.gitlab.arturbosch.detekt.api.internal` package is only used in the detekt-api project so there shouldn't be any need to ignore it in detekt-psi-utils. `:detekt-psi-utils:apiCheck` & `:detekt-api:apiCheck` tasks both run successfully with this config.

**Pull request merged**